src/MatrixFields/field_matrix_solver.jl

```diff
@@ -247,6 +247,10 @@ function check_field_matrix_solver(::BlockDiagonalSolve, _, A, b)
     end
 end
 
+cheap_inv(_) = false
```

Comment:
What's an example where this will be false? I thought UniformScaling and ColumnwiseBandMatrixField were the only two options for a FieldMatrix.

Reply:
So, maybe we can just change `cheap_inv(::ColumnwiseBandMatrixField) = true` to `cheap_inv(A::ColumnwiseBandMatrixField) = eltype(A) <: DiagonalMatrixRow`? I think this is what I had in mind.

Reply:
Yeah, I think that is the correct one, thanks for catching that @dennisYatunin!

Reply:
Okay, limiting this to diagonal matrices makes sense! And I agree that the dispatch is getting pretty complicated with all the single/multiple field solve performance specializations; it would be good to abstract out this logic somehow.